include/zephyr/net/net_if.h

```diff
@@ -857,10 +857,6 @@ static inline int net_if_set_link_addr_unlocked(struct net_if *iface,
 						uint8_t *addr, uint8_t len,
 						enum net_link_type type)
 {
-	if (net_if_flag_is_set(iface, NET_IF_UP)) {
-		return -EPERM;
-	}
-
```

Comment:
I am not sure about this. Don't we need some check here so that it is not possible to set the address if the interface is up and running as that would confuse the IPv6 stack? Perhaps something like
```
	if (net_if_flag_is_set(iface, NET_IF_RUNNING)) {
		return -EPERM;
	}
```
WDYT?

Reply:
I need to investigate 802.15.4 a bit, I've tried to leave the check as suggested, but there's some hacky code in 802.15.4 L2 causing some tests to fail:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/l2/ieee802154/ieee802154_mgmt.c#L206

Worst case I'll need to update the flag in 802.15.4 L2 as well..

Reply:
It seems that there is a case for IEEE802154 when the LL address can be updated during interface operation (processing MAC command, updating to short address). Therefore for now, I've updated the workaround which allowed to do this by using `NET_IF_RUNNING` flag instead of `NET_IF_UP`.